server/controllers/users.js

```diff
@@ -190,20 +191,31 @@ export function getUserCampaignAssociation(req, res) {
     });
 }
 
-export function updateUserCallSIDField(req, res) {
-  const { id } = req.params;
-  const call_sid = 'CAdksl234591adfide294821kdau3u3933';
-  const params = { id, call_sid };
-  return usersService.updateUserById(params)
-    .then((user) => {
-      if (user) {
-        res.status(200).json(user);
-      } else {
-        res.status(404).json({ message: 'Could not process request to update user Call SID' });
-      }
+export function startTwilioConnection(req, res) {
+  const { id, campaign_id } = req.params;
+  usersService.getUserById({ id })
+    .then((volunteer) => {
+      const { phone_number } = volunteer.attributes;
+      callVolunteer(id, campaign_id, phone_number)
+        .then((call) => {
+          const { sid: call_sid } = call;
+          const params = { id, call_sid };
+          return usersService.updateUserById(params)
+            .then((user) => {
+              if (user) {
+                res.status(200).json(user);
+              } else {
+                res.status(404).json({ message: 'Could not process request to update user Call SID' });
+              }
+            })
+            .catch((err) => {
+              res.status(500).json({ message: `Could not process request to update user Call SID: ${err}` });
+            });
+        })
+        .catch(err => console.log(err));
```

Comment:
Nice error handling all around: how about we also return a 500 here with an appropriate client facing message? And also log the error in its entirety here as you're doing already?

Reply:
Good catch. Will do!